pylint_protobuf/transform.py

```diff
@@ -105,7 +112,20 @@ def _template_message(desc, depth=0):
         '    self.{} = {}()\n'.format(field_name, field_type)
         for field_name, field_type in external_fields
     )
-    helpers = 'def __getitem__(self, idx):\n    pass\n'
+    helpers = ""
+    if desc.has_options and desc.GetOptions().map_entry:
+        # for map <key, value> fields
+        # This mirrors the _IsMessageMapField check
+        value_type = desc.fields_by_name['value']
+        if value_type.cpp_type == FieldDescriptor.CPPTYPE_MESSAGE:
```

Comment:
I see that this is exactly how it goes in `_IsMessageMapField` but I remember dealing with this and I think my approach was to branch on `value_type.type == FieldDescriptor.TYPE_MESSAGE`. I wonder why they chose to branch on `cpp_type` specifically? The `FieldDescriptor` for the `*Entry` types seems to set `type=11` as expected.